docs/features/techdocs/how-to-guides.md

```diff
@@ -142,6 +142,78 @@ const AppRoutes = () => {
 };
 ```
 
+## How to customize the TechDocs reader page?
+
+### Custom Header
+
+This is done in your `app` package. By default, you might see something like
+this in your `App.tsx`:
+
+```tsx
+const AppRoutes = () => {
+  <FlatRoutes>
+    <Route
+      path="/docs/:namespace/:kind/:name/*"
+      element={<TechDocsReaderPage />}
+    />
+  </FlatRoutes>;
+};
+```
+
+But the `<TechDocsReaderPage />` component accepts children, which means that
+you can customize the page with any other Header component you would like to
+use.
+
+Most likely, you would want to create and maintain such a component in a new
+directory at `packages/app/src/components/techdocs`, and import and use it in
+`App.tsx`:
+
+```tsx
+<Route
+  path="/docs/:namespace/:kind/:name/*"
+  element={
+    <TechDocsReaderPage>
+      {({ techdocsMetadataValue, entityMetadataValue, entityId }) => (
```

Comment:
Hmm...!  So the proposal here is to make the header customizable by passing it as a child to the routable extension?

Thinking through how we do this in other composable bits in Backstage.

- In the Home plugin, we ask integrators to put this [within the composable outlet itself](https://github.com/backstage/backstage/blob/master/packages/app/src/components/home/HomePage.tsx#L35).
- In the Search plugin, we also require this in the [composable outlet](https://github.com/backstage/backstage/blob/master/packages/app/src/components/search/SearchPage.tsx#L49)
- In the catalog entity page, it's a little more complicated (because it's [hard-coded in the EntityLayout](https://github.com/backstage/backstage/blob/master/plugins/catalog/src/components/EntityLayout/EntityLayout.tsx#L225)), but I suppose one could just use a different layout wrapper and customize the header.

I can imagine scenarios where a TechDocs adopter would want to add extra stuff to the reader page itself, in addition to just the header (say, a row of extra metadata between the header and the reader itself).  Would it make sense to follow the same pattern here?

Reply:
Yeah, when starting to implement these things, I came to the same point, that its probably a need for more customization than just the Header of the Reader page. I decided to open the PR and get your thoughts (as our need specifically was for only the Header now) so I'm happy you raised it here. I think either it can be something to iterate on OR we can make the entire reader page customizable directly to follow the same patterns as other composable thing in Backstage, what do you think?

Reply:
+1